articles/becoming-a-flutter-developer/index.md

```diff
@@ -0,0 +1,179 @@
+
+Here are some of the most popular design tools:
+- Sketch.
+- Adobe XD.
+- [Figma.](https://www.figma.com/) 
+- [Flutter Studio.](https://flutterstudio.app/)
+
+After designing, you can use [Supernova](https://www.supernova.io/) to convert the design into code by exporting it. You will, however, need to refactor the exported code to break it into readable code.
+
+### Flutter Developer Skills
+
+1. **Dart Programming Language**
+
+`Dart` is a client-based programming language that is scalable and can write simple scripts or full-featured applications. It is an object-oriented and dynamic language. Google developed dart. It is the primary language used in Flutter.
+
+2. **Knowledge on Agile Methodology and Waterfall for Software Development**
+
+A software development method is dividing the software development process into different phases. It enhances product design and resource utilization ***for example*** money and time. 
+
+The most commonly used methodologies are `Waterfall Methodology` and `Agile Methodology`. A Flutter developer needs to familiarize themselves with the software methodologies to work with ease in a software development team.
+
+3. **Prior knowledge on a Native Language**
+
+The native languages for mobile application development include:
+
+1. `Swift` for iOS applications.
+
+2. `Java` and `Kotlin` for Android applications.
+
+Understanding these languages will make it easier to become a flutter developer. Learning Dart with the knowledge of a native language makes it easier.
+
+
+### Mistakes to Avoid When Learning Flutter Development
+
+- **Learn native languages, not just flutter and dart.**
+
+Most software developers will argue that prior knowledge in a native language is unnecessary, but it is a skill requirement on most job descriptions. Flutter development is still fairly new, and employers must have proof that you know about application development. The only way to show this is by having experience with native languages were the only ones used to develop most commercial and large-scale applications.
+
+Knowing a native language will give you a groundbreaking compared to a person who only knows flutter and dart. It will display a variety of expertise and knowledge. A grasp on native languages will make it easier to move flutter development.
+
+- **Do not over-rely on one source to develop your career as a Flutter Developer.**
+
+Flutter and Dart are still fairly new and not fully adapted. Finding a tutorial or reading material with a sufficient amount of knowledge to kick off your career might prove to be a bit hard. It is advisable to look into multiple resources to get a good grasp of the required knowledge. Different tutors have unique experiences and knowledge of the framework and programming language.
```

Comment:
I believe Flutter is fully adapted now. They released their `beta` releases some years ago. Now, they release `stable` versions or updates regularly. Check out companies using it https://flutter.dev/showcase

Reply:
@WanjaMIKE I did not mean that it is not being used by companies, of course it is. I meant the knowledge and level of understanding of the framework is not sufficient to make a learner rely on one source, most questions even on stack overflow are unanswered, its about 3-4 years old, with a slow adaption rate. or what did you mean?

Reply:
Okay. You can rephrase the sentence then. Thanks